crates/consensus/beacon/src/engine/invalid_headers.rs

```diff
@@ -56,15 +66,56 @@ impl InvalidHeaderCache {
 
     /// Inserts an invalid ancestor into the map.
     pub(crate) fn insert(&mut self, invalid_ancestor: SealedHeader) {
-        if self.headers.get(&invalid_ancestor.hash).is_none() {
+        if self.get(&invalid_ancestor.hash).is_none() {
             let hash = invalid_ancestor.hash;
             let header = invalid_ancestor.unseal();
             warn!(target: "consensus::engine", ?hash, ?header, "Bad block with hash");
-            self.headers.insert(hash, Arc::new(header));
+            self.insert_entry(hash, Arc::new(header));
 
             // update metrics
             self.metrics.unique_inserts.increment(1);
             self.metrics.invalid_headers.set(self.headers.len() as f64);
         }
     }
 }
+
+struct HeaderEntry {
+    /// Keeps track how many times this header has been hit.
+    hit_count: u8,
+    /// The actually header entry
+    header: Arc<Header>,
+}
+
+/// Metrics for the invalid headers cache.
+#[derive(Metrics)]
+#[metrics(scope = "invalid_header_cache")]
```

Comment:
We should namespace this like other metrics based on what component it is part of, `reth_invalid_header_cache` doesn't really say much

Reply:
agree, will do that separately.